src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -3006,14 +3006,13 @@ else if (column >= getColumnCount()) {
         }
         else {
             TableColumnModel cm = getColumnModel();
-            if( getComponentOrientation().isLeftToRight() ) {
-                for(int i = 0; i < column; i++) {
-                    r.x += cm.getColumn(i).getWidth();
-                }
-            } else {
-                for(int i = cm.getColumnCount()-1; i > column; i--) {
-                    r.x += cm.getColumn(i).getWidth();
-                }
+            for(int i = 0; i < column; i++) {
+                r.x += cm.getColumn(i).getWidth();
+            }
+            //x position is updated to handled the offset to which the table
+            //has to be moved towards right side in right-to-left Orientation.
+            if( !getComponentOrientation().isLeftToRight() ) {
+                r.x = getWidthInRightToLeft() - r.x - cm.getColumn(column).getWidth();
             }
             r.width = cm.getColumn(column).getWidth();
```

Comment:
```suggestion
            final int columnWidth = cm.getColumn(column).getWidth();
            if( !getComponentOrientation().isLeftToRight() ) {
                r.x = getWidthInRightToLeft() - r.x - columnWidth;
            }
            r.width = columnWidth;
```
Just to avoid getting the same value twice.